src/functions-and-arrays.js

```diff
@@ -106,10 +241,44 @@ const matrix = [
   [1, 70, 54, 71, 83, 51, 54, 69, 16, 92, 33, 48, 61, 43, 52, 1, 89, 19, 67, 48]
 ];
 
-function greatestProduct() {}
-
-
+function greatestProduct(arr) {
+  let rows = arr.length;
+  let cols = arr[0].length;
+  let horizontal = 1;
+  let greatestHorizontal = 0;
+  let vertical = 1;
+  let greatestVertical = 0;
+
+  for (i = 0; i < rows; i++) {
```

Comment:
Não esqueça o `let` para declarar a variável `i`.
Para q o `i` não fuja do escopo e ser acessado de fora do `for`, isso é, ele tem que "nascer e morrer" dentro do `for` apenas.